core/src/main/java/io/parsingdata/metal/expression/value/CoreValue.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2013-2016 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.expression.value;
+
+import static io.parsingdata.metal.Util.bytesToHexString;
+import static io.parsingdata.metal.Util.checkNotNull;
+import static io.parsingdata.metal.encoding.Encoding.DEFAULT_ENCODING;
+
+import java.math.BigInteger;
+import java.util.BitSet;
+import java.util.Objects;
+
+import io.parsingdata.metal.Util;
+import io.parsingdata.metal.data.Slice;
+import io.parsingdata.metal.encoding.ByteOrder;
+import io.parsingdata.metal.encoding.Encoding;
+import io.parsingdata.metal.encoding.Sign;
+
+public class CoreValue implements Value {
```

Comment:
I chose this name because it contains the _core_ functionality of all regular `Value` implementations. Not just the constructed values used it, but `ParseValue` derives from it as well. So that's what I was aiming at with the "core" part.

I do agree it's difficult to find a good name. The only other one I could come up with was `ValueImpl` (which is normally ok, but not for something that actually gets instantiated like this). I'm open to changing names if we discover something that really fits well.

Reply:
DIfficult. `AnyValue`? `JustAValue`?

Reply:
Let's revisit this if the name really bothers us going forward :)